tests/conftest.py

```diff
@@ -238,3 +257,79 @@ def model_to_delete(self, models_with_kwargs):
+        """Make a fixture for use with ``test_database.CreateLogic.test_create``.
+
+        :param model_fixture: The ``ModelFixture`` object. Note that the generic fixtures (defined
+          in ``CreateFixtures``) use a lazy fixture (``models_with_kwargs``) which represents a
+          parametrization of all models in the ``pangeo_forge_orchestrator.models.MODELS`` dict.
+          By contrast, fixture classes which inherit from this mixin will pass only the single,
+          specific ``ModelFixture`` representing the database model in which the targeted
+          categorical field exists.
+        :param request: The pytest request object containing a parametrization of all valid options
+          for the categorical field. E.g.: ``@pytest.fixture(params=[opt.value for opt in Obj])``,
+          where ``Obj`` is the ``enum`` object that defines the categorical options to test.
+        """
+
+        kw_0, _ = model_fixture.kwargs
+        kw_0.request[self.field_name] = request.param
+        return model_fixture.models, kw_0.request, kw_0.blank_opts
+
+    def make_failing_model_to_create(
+        self, model_fixture: ModelFixture
+    ) -> Tuple[MultipleModels, dict, str]:
+        """Make a fixture for use with ``test_database.CreateLogic.test_create_failure``.
+
+        :param model_fixture: The ``ModelFixture`` object. See docstring for
+          ``make_model_to_create`` method on this class for full description.
+        """
+        kw_0, _ = model_fixture.kwargs
+        failing_request = copy.deepcopy(kw_0.request)
+        failing_request[self.field_name] = self.invalid_value
+        return model_fixture.models, failing_request, self.failure_mode
+
+
+class RecipeRunStatusFixtures(EnumFixtureFactory):
+    """Fixtures for testing validation of values passed to recipe run table's `status` field.
+    """
+
+    field_name = "status"
+    invalid_value = "invalid_status"
+
+    @pytest.fixture(params=[opt.value for opt in RecipeRunStatus])
+    def model_to_create(self, recipe_run_with_kwargs: ModelFixture, request):
+        return self.make_model_to_create(recipe_run_with_kwargs, request)
+
+    @pytest.fixture
+    def failing_model_to_create(self, recipe_run_with_kwargs: ModelFixture):
+        return self.make_failing_model_to_create(recipe_run_with_kwargs)
+
+
+class RecipeRunConclusionFixtures(EnumFixtureFactory):
+    """Fixtures for testing validation of values passed to recipe run table's `conclusion` field.
+    """
+
+    field_name = "conclusion"
+    invalid_value = "invalid_conclusion"
+
+    @pytest.fixture(params=[opt.value for opt in RecipeRunConclusion])
+    def model_to_create(self, recipe_run_with_kwargs: ModelFixture, request):
+        return self.make_model_to_create(recipe_run_with_kwargs, request)
+
+    @pytest.fixture
+    def failing_model_to_create(self, recipe_run_with_kwargs: ModelFixture):
+        return self.make_failing_model_to_create(recipe_run_with_kwargs)
```

Comment:
These sub-classes are a bit duplicative. Could you think of a way to roll this logic into the parent class (`EnumFixtureFactory`) so that you only have to write

```python
class RecipeRunConclusionFixtures(EnumFixtureFactory):
    field_name = "conclusion"
    invalid_value = "invalid_conclusion"
    enum_class = RecipeRunConclusion
```

Reply:
TL;DR I pretty sure I've found a way to refactor conftest.py to make it more concise while also lifting these type validation tests up into the primary set of CRUD tests. (As opposed to their current position: tacked onto the end of the primary tests.) Ultimately, we'll want to do a lot more type validation testing and my initial approach here of defining dedicated objects for each type doesn't seem sustainable. Commits to follow.

Slightly longer version: I was struggling yesterday with the fact that pytest test classes [cannot have a constructor](https://stackoverflow.com/a/21431187) (see also [here](https://stackoverflow.com/a/58688529)). Because the `Test*` classes inherit from these fixture classes, pytest complains if we try to dynamically assign fixtures via an `__init__` (or `__post_init__`, etc.) method in `EnumFixtureFactory`. Interestingly, [this approach of dynamic fixture generation](https://github.com/pytest-dev/pytest/issues/2424#issuecomment-333387206) in which `globals()` is directly mutated seems to receive the approval of some of pytest's core maintainers. I experimented with that a bit and it does work, but it feels better to avoid if possible. The forthcoming conftest.py refactor commits solve the underlying issue (conciseness of type validation test fixtures) in a more standard style.